### DISCUSSION TITLE
docs: select from subquery using Query Builder

### DISCUSSION THREAD

**Discussion by alex996** [mikro-orm/mikro-orm#3208] [Q&A] [answered]
Is it possible to write the following query with Query Builder?

```sql
select a, b, a + b from (
  select sum(col1) as a, sum(col2) as b from some_table
) q;
```

The [docs on sub-queries](https://mikro-orm.io/docs/query-builder#using-sub-queries) don't mention this use case unfortunately. Knex [appears to support it](https://github.com/knex/knex/issues/2504#issuecomment-367445831), but I can't quite figure out the syntax that works in Mikro:

```ts
const qb = orm.em.createQueryBuilder(MyEntity);
const knex = qb.getKnexQuery();

// attempt #1
const s = knex.select('a, b, a + b').from(/* what goes here? */)
// attempt #2
const s = knex.select('a, b, a + b').fromRaw() // Property 'fromRaw' does not exist on type
// attempt #3
const s = knex.with('sq', knex.raw()) // Property 'raw' does not exist on type
```

Is there any way to do it besides writing a raw query by hand?

Thank you.

**Comment by B4nan** [comment, marked as answer]
Not supported with the ORM query builder, its mainly for working with entities, not that much with random data. If you dont know how to do it with knex, ask in the knex GH or discord, that part is not connected to the ORM anyhow. 

Hint 1 - you need two (knex) QB instances for your use case.
Hint 2 - your `knex` object is not the same as the one you see in knex docs - it is already configured QB instance that will by default produce a `select * from entity` query.
Hint 3 - use `em.getKnex()` if you want the general knex instance, not a QB bound to some entity
